model_inference/inference.py

```diff
@@ -31,26 +31,44 @@ async def process_inference_results(data, imageDims):
         for i, box in enumerate(data[0]["boxes"]):
             for j, box2 in enumerate(data[0]["boxes"]):
                 if j > i:
+
```

Comment:
Correct me if I'm wrong, but from my understanding in the old version, the overlapping condition was based on the relative positions of the boxes. If one box was entirely within the other, it was considered overlapping.

In the new version, the overlapping condition is based on the area of intersection between two boxes. If the common area is more than half of either box, they are considered overlapping. This is a more accurate way to determine overlap, as it considers partial overlaps as well.

```suggestion
I think t would be helpful to explain the new overlapping condition and why it's more accurate.
```

Reply:
Before if part of a box was inside the other, the two boxes were considered overlapping. Even if their common area was tiny.

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/c0446dc1-9b23-4dc6-8ed7-a17cd15e6dd1)

For example in the image the two boxes are overlapping, meaning that the application would display the image like this:

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/0f0f2029-6934-41d1-b370-7cfb4adc04df)

But the overlap here is very tiny and (in my humble opinion) doesn't impair the user to recognize which box belongs to which seed. However, in case the boxes have a common area way bigger, like in the next image, where the two boxes are basically the same seed, we don't want to display two boxes.

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/156c7a0b-21ed-4bad-b7e9-d50c5312ac68)

One is enough:

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/08a195cb-9371-4c0a-9de1-f9531c5f3c0b)

The mark bigger than 50% of the area of both boxes is arbitrary and could be a parameter in the function that is set by the user (or the frontend dev) to give them ownership of how they display the boxes.

Hope this sums it up. I will update the docstring of the function to better explain the change as you point it out.

Reply:
Done